base/channels.jl

```diff
@@ -457,3 +457,24 @@ function iterate(c::Channel, state=nothing)
 end
 
 IteratorSize(::Type{<:Channel}) = SizeUnknown()
+
+"""
+    threaded_foreach(f, channel::Channel; ntasks=Threads.nthreads(), async=true)
+
+Similar to `foreach(f, channel)`, but iteration over `channel` and calls to
+`f` are split across `ntasks` Tasks spawned by `Threads.@spawn`.
+
+If `async` is `false`, this function will `wait` for all internally spawned tasks
+to complete before returning.
+"""
+function threaded_foreach(f, channel::Channel; ntasks=Threads.nthreads(), async=true)
+    tasks = map(1:ntasks) do _
+        Threads.@spawn begin
+            for item in channel
+                wait(Threads.@spawn f(item))
```

Comment:
Maybe a bit too tangent, but FYI I noticed you can get a bit more speedup https://github.com/tkf/Transducers.jl/pull/123#issuecomment-579010628 if you collect the elements into a buffer `Vector` and then iterate over it. You may want to improve latency instead of throughput so I guess it's not always a better choice. But I wonder if it makes sense to add a "buffer size" option to it.

Reply:
This indeed sounds like a useful option! I'd probably want to fiddle around with it in a follow-up PR, though.